tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_interactive_view.ts

```diff
@@ -196,17 +200,47 @@ export class LineChartInteractiveViewComponent
         this.changeDetector.markForCheck();
       });
 
+    fromEvent<MouseEvent>(window, 'keydown', {
+      passive: true,
+    })
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((event) => {
+        const newState = this.shouldPan(event);
+        if (newState !== this.specialKeyPressed) {
+          this.specialKeyPressed = newState;
+          this.changeDetector.markForCheck();
+        }
+      });
+
+    fromEvent<MouseEvent>(window, 'keyup', {
+      passive: true,
+    })
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((event) => {
+        const newState = this.shouldPan(event);
+        if (newState !== this.specialKeyPressed) {
+          this.specialKeyPressed = newState;
+          this.changeDetector.markForCheck();
+        }
+      });
+
     fromEvent<MouseEvent>(this.dotsContainer.nativeElement, 'mousedown', {
       passive: true,
     })
       .pipe(takeUntil(this.ngUnsubscribe))
       .subscribe((event) => {
-        this.state = event.shiftKey
+        const prevState = this.state;
+        const nextState = this.shouldPan(event)
           ? InteractionState.PANNING
           : InteractionState.DRAG_ZOOMING;
-        this.dragStartCoord = {x: event.offsetX, y: event.offsetY};
 
-        if (this.state === InteractionState.DRAG_ZOOMING) {
+        // In case left down -> middle down, don't override the initial
```

Comment:
Perhaps reword as "Override values X and Y except for the case ZZZ"? Maybe you could explain what prevState and nextState would be for case ZZZ?